### DISCUSSION TITLE
updating previous content to HasDrafts

### DISCUSSION THREAD

**Discussion by retosteffen** [oddvalue/laravel-drafts#30] [Q&A] [answered]
I have content already in the database and would like to add laravel-drafts but I can't see a way to update the old content.
There is the ->publish()->save() that publishes the model but doesn't fill out the uuid, publisher_type, publisher_id info.
It would be great to have a way to add that to existing content in the database.

**Comment by oddvalue** [comment, marked as answer]
Hi @retosteffen,

Thanks for the suggestion. It's a bit tricky to do since there's no way of knowing what the publisher should be automatically. Your best bet is to set the data manually in the database if you know what you want the values to be. 
Regarding the uuid, you can fill them in for existing models like this:

```php
Model::query()->whereNull('uuid')->each(function (Model $record): void {
    $record->generateUuid();
    $record->save();
});
```
